src/utils/database-common-macros/src/lib.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::parse::{Parse, ParseStream};
+use syn::{parse_macro_input, FnArg, Ident, ItemFn, Token, Type};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[proc_macro_attribute]
+pub fn transactional_handler(_attr: TokenStream, item: TokenStream) -> TokenStream {
+    const CATALOG_ARGUMENT: &str = "Extension(catalog): Extension<Catalog>";
```

Comment:
would be nice if this was resistant to variations like:
```rust
Extension(catalog): Extension<Catalog>
extensions::Extension(catalog): extensions::Extension<Catalog>
Extension(catalog): Extension<dill::Catalog>
```

Reply:
I thought about it, since my original implementation looked like this:
```rust
::axum::Extension(catalog): ::axum::Extension<::dill::Catalog>
```

```
Pros
+ No additional imports needed
Сons
- Ugly
```

Then I tried covering the different combinations programmatically -- the code started to turn into a nightmare.

---

But I'll think about what I can do -- I think if we only take the last pieces of identifiers, it might work

Reply:
Done